kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/OperatorMain.java

```diff
@@ -66,28 +80,85 @@ void start() {
         operator.installShutdownHook(Duration.ofSeconds(10));
         var registeredController = operator.register(new ProxyReconciler(runtimeDecl()));
         // TODO couple the health of the registeredController to the operator's HTTP healthchecks
+        managementServer.createContext("/", exchange -> {
+            try (exchange) {
+                // note while the JDK docs advise exchange.getRequestBody().transferTo(OutputStream.nullOutputStream()); we explicitly don't do that!
+                // As a denial-of-service protection we don't expect anything other than GET requests so there should be no input to read.
+                exchange.sendResponseHeaders(404, -1);
+            }
+        }).getFilters().add(UnsupportedHttpMethodFilter.INSTANCE); // This works as a request to `/metrics` has a more explicit match and thus gets served.
+        managementServer.start();
         operator.start();
         LOGGER.info("Operator started.");
     }
 
     void stop() {
         operator.stop();
+        managementServer.stop(0); // TODO maybe this should be configurable
         LOGGER.info("Operator stopped.");
     }
 
     @NonNull
     static RuntimeDecl runtimeDecl() {
         // TODO read these from some configuration CR
-        return new RuntimeDecl(List.of(
-                new FilterApiDecl("filter.kroxylicious.io", "v1alpha1", "KafkaProtocolFilter")));
+        return new RuntimeDecl(List.of(new FilterApiDecl("filter.kroxylicious.io", "v1alpha1", "KafkaProtocolFilter")));
     }
 
     private MicrometerMetrics enablePrometheusMetrics() {
-        registry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
-        Metrics.globalRegistry.add(registry);
         return MicrometerMetrics.newPerResourceCollectingMicrometerMetricsBuilder(Metrics.globalRegistry)
                 .withCleanUpDelayInSeconds(35)
                 .withCleaningThreadNumber(1)
                 .build();
     }
+
+    private void configurePrometheusMetrics(HttpServer managementServer) {
+        final PrometheusMeterRegistry prometheusMeterRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+        final HttpContext metricsContext = managementServer.createContext("/metrics",
+                new MetricsHandler(prometheusMeterRegistry.getPrometheusRegistry()));
+        metricsContext.getFilters().add(UnsupportedHttpMethodFilter.INSTANCE);
+        Metrics.globalRegistry.add(prometheusMeterRegistry);
+    }
+
+    @VisibleForTesting
+    static HttpServer createHttpServer() throws IOException {
+        final Properties systemProps = System.getProperties();
+        if (!systemProps.containsKey("sun.net.httpserver.maxReqTime")) {
+            System.setProperty("sun.net.httpserver.maxReqTime", "60");
+        }
+
+        if (!systemProps.containsKey("sun.net.httpserver.maxRspTime")) {
+            System.setProperty("sun.net.httpserver.maxRspTime", "120");
```

Comment:
We should maybe support overriding this from an ENV Var for better support in a k8s env. Thoughts? Maybe we should leave that until we see an actual need. As two mins seems rather generous anyway.